public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+  because `HeroPanel` is a child of `HeroJobBoard`. However, we need to find how to inject `MessageBus` into `MessageBoard` that is totally unrelated to
+  the other two components.
+  
+  We should make `HeroJobBoard` and `MessageBoard` to be two direct children of their hosting component. In the sample code of this chapter, we use routing that
+  allows us to host all chapter scenarios in a single application: each scenario is located in a different routing path, and they are rendered through the 
+  `router-outlet` directive.
+  
+  To demonstrate that `MessageBoard` is unrelated to `HeroJobBoard` &mdash; it is in a separate branch of the component hierarchy &mdash; we create
+  `JobBoardWithMessages`, merely to embed these two components:
+  
++makeExample('component-communication/ts/app/message-board/job-board-with-messages.ts', null, 'message-board/message-board.ts', {blk: /(providers: \[MessageBus\])/g})
+
+:marked
+.alert.is-helpful
+  :marked
+    Of course, we set up the routing configuration for this scenario path so that the `JobBoardWithMessages` is used as the root component displayed by
+    `router-outlet`.
+  
+:marked
+  This code shows the simple solution to inject the same instance of `MessageBus` into all components: we use the `providers` annotation property of
+  `JobBoardWithMessages`. Because `MessageBoard` and `HeroJobBoard` are both the children of `JobBoardMessage`, they will be injected with the 
+  same `MessageBus` instance. When instantiating `HeroPanel`, the framework will use the `MessageBus` instance injected into `HeroJobBoard`, 
+  because of the parent-child relationship between `HeroJobBoard` and `HeroPanel`.
+  
+.alert.is-important
+  :marked
+    We must not include `MessageBus` in the `providers` of `HeroJobBoard`, because it would break the logic. In that case `MessageBoard` would listen
+    to a separate `MessageBus` the other components are talking to. This is how the correct setup of `HeroJobBoard` is:
+
++makeExample('component-communication/ts/fragments/no-message-bus-in-hero-job-board.ts', null, 'message-board/hero-job-board.ts (extract)', {blk: /(providers: \[JobService\])/g})
+
+:marked
+  Now, we put together all pieces to check how the message bus works:
+  
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/message-board-ui.png" alt="MessageBoard in action")
+  
+:marked
+  ## #6: Parent to child communication where children are transposed in &mdash; using `@ContentChildren`
+  
+  There are situations, when a parent component cannot determine the set of its child components on its own, because the collection of its children is managed
+  by an external entity. However, even in this case, the parent needs to communicate with each children. How can the parent establish the communication channel 
+  to its children? Angular provides several mechanism to achieve this, here we show the usage of the `@ContentChildren` annotation.
```

Comment:
Several mechanisms?  What are they other than the content children example you're about to look at?  Why would I pick this mechanism over the others?

Btw, the only other mechanism that comes to mind is the shared service mentioned earlier. Why choose this approach over that one?